File: solutions/0648-replace-words/solution.py
class Solution(object):
    def replaceWords(self, dictionary, sentence):
        """
        :type dictionary: List[str]
        :type sentence: str
        :rtype: str
        """
        ls=sentence.split()
        n=len(ls)
        for i in dictionary:
            for j in range(n):
                if i in ls[j][:len(i)]:
                    ls[j]=i
        return " ".join(ls)

File: solutions/0648-replace-words/solution.java
class Solution {
    public String replaceWords(List<String> dictionary, String sentence) {
        String[] ls=sentence.split("\\ ");
        int n=ls.length;
        for(String i:dictionary)
        {
            for(int j=0;j<n;j++)
            {
                if(ls[j].startsWith(i))
                    ls[j]=i;
            }
        }
        return String.join(" ",ls);
    }
}
